### PR TITLE
Naterose/dev audio

### PR DESCRIPTION
**Pull request by NathanielRose**
adding customized inference script with normalized json output for keras and tensorflow frameworks.

### CODE REVIEW

File: microsoft/validation/inference_js.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+import time
+import numpy
+import tensorflow as tf
+from tensorflow.python.lib.io import file_io
+from tensorflow import app
+from tensorflow import flags
+from tensorflow import gfile
+from tensorflow import logging
+
+import eval_util
```
Comment (evanlouie) [inline microsoft/validation/inference_js.py]: Can the script run from this directory? or does it need to be moved into `youtube8m`?

File: microsoft/validation/k_inference_js.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+from keras.activations import relu, softmax
+from keras.callbacks import (EarlyStopping, LearningRateScheduler,
+                             ModelCheckpoint, TensorBoard, ReduceLROnPlateau)
+from keras.layers import (Convolution2D, GlobalAveragePooling2D, BatchNormalization, Flatten,
+                          GlobalMaxPool2D, MaxPool2D, concatenate, Activation, Input, Dense)
+from keras.utils import Sequence, to_categorical
+from keras import backend as K
+import argparse
+
+parser = argparse.ArgumentParser(
```
Comment (evanlouie) [inline microsoft/validation/k_inference_js.py]: we should be wrapping the executable portion of scripts in a

`if __name__ == "__main__":`

File: microsoft/validation/inference_js.py (hunk 1 above)
Comment (NathanielRose) [reply microsoft/validation/inference_js.py]: No needs to be in youtube8m. Kept it here since the Keras inference isnt dependent on yt8m. What do you think?

File: microsoft/validation/k_inference_js.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    model = models.Model(inputs=inp, outputs=out)
+    opt = optimizers.Adam(config.learning_rate)
+
+    model.compile(
+        optimizer=opt, loss=losses.categorical_crossentropy, metrics=['acc'])
+    return model
+
+
+config = Config(sampling_rate=44100, audio_duration=2, n_folds=10,
```
Comment (evanlouie) [inline microsoft/validation/k_inference_js.py]: All of this should be in the `if __name__ == "__main__":` as well. Pull down the `argparser` stuff down to here and in put it all in one big `if __name__ == "__main__":`

File: microsoft/validation/inference_js.py (hunk 1 above)
Comment (evanlouie) [reply microsoft/validation/inference_js.py]: It's not clean to have it in this directory, but I also don't like separating this script from the Keras one :(
For now, could you just add some documentation to the top of the file noting that it needs the yt8m code to run?

Comment (yradsmikham) [reply microsoft/validation/inference_js.py]: let's also rename this back to "inference_json.py". Let's also have the "main" function include a return(json) so that it displays the json output. I'm happy to make this change once it is merged as well.

Comment (NathanielRose) [reply microsoft/validation/inference_js.py]: Added readme @yradsmikham Ill look to you to add the edit the script for what you need.

File: microsoft/validation/k_inference_js.py (hunk 3 above)
Comment (NathanielRose) [reply microsoft/validation/k_inference_js.py]: Working!
